src/main/java/com/scalyr/integrations/kafka/ScalyrSinkTask.java

```diff
@@ -166,7 +166,6 @@ public void put(Collection<SinkRecord> records) {
       if (noRecordLogRateLimiter.tryAcquire()) {
```

Comment:
I would also add a comment here explicitly saying you are not returning early because we use this opportunity to see if `batchSendWaitMs` has been exceeded.